src/Holding/ViewHolding/HoldingBoundWith/HoldingBoundWith.js

```diff
@@ -0,0 +1,77 @@
+import PropTypes from 'prop-types';
+import { get } from 'lodash';
+import { FormattedMessage } from 'react-intl';
+import { Link } from 'react-router-dom';
+
+import {
+  Accordion,
+  MultiColumnList,
+} from '@folio/stripes/components';
+
+import { IntlConsumer } from '@folio/stripes/core';
+import { noValue } from '../../../constants';
+import { checkIfArrayIsEmpty } from '../../../utils';
+import useBoundWithHoldings from './useBoundWithHoldings';
+
+const HoldingBoundWith = ({ boundWithItems }) => {
+  const { isLoading, boundWithHoldings } = useBoundWithHoldings(boundWithItems);
+  const data = boundWithItems.records?.map(boundWithItem => ({
+    item: boundWithItem,
+    holdingsRecord: boundWithHoldings.find(
```

Comment:
@maccabeelevine this could be simplified by first converting `boundWithHoldings` into a map via `keyBy` from `lodash`:

````js
const boundWithHoldingsMapById = keyBy(boundWithHoldings, 'id');
````

with that, you could turn it into:

````js
const data = boundWithItems.records?.map(boundWithItem => ({
  item: boundWithItem,
  holdings record: boundWithHoldings[boundWithItem.holdingsRecordId]
});
````
That should be a bit more performant since you won't need to loop via `boundWithHoldings` for every `boundWithItem`.

Reply:
Good catch, I've made this change.